analyzers/analysistest/analysistest.go

```diff
@@ -0,0 +1,269 @@
+	"github.com/smacker/go-tree-sitter/java"
+	"github.com/smacker/go-tree-sitter/javascript"
+	"github.com/smacker/go-tree-sitter/lua"
+	"github.com/smacker/go-tree-sitter/ocaml"
+	"github.com/smacker/go-tree-sitter/php"
+	"github.com/smacker/go-tree-sitter/protobuf"
+	"github.com/smacker/go-tree-sitter/python"
+	"github.com/smacker/go-tree-sitter/ruby"
+	"github.com/smacker/go-tree-sitter/rust"
+	"github.com/smacker/go-tree-sitter/scala"
+	"github.com/smacker/go-tree-sitter/svelte"
+	"github.com/smacker/go-tree-sitter/toml"
+	"github.com/smacker/go-tree-sitter/typescript/typescript"
+	"github.com/smacker/go-tree-sitter/yaml"
+)
+
+// ParsedIssue represents an issue parsed using tree-sitter.
+type ParsedIssue struct {
+	IssueCode string
+	Line      int
+}
+
+type ParsedIssues []ParsedIssue
+
+func Run(directory string) error {
+	// read the generated report from TOOLBOX_PATH
+	toolboxPath := os.Getenv("TOOLBOX_PATH")
+	generatedFile := path.Join(toolboxPath, "analysis_report.json")
+	reportContent, err := os.ReadFile(generatedFile)
+	if err != nil {
+		return err
+	}
+
+	var report types.AnalysisReport
+	err = json.Unmarshal(reportContent, &report)
+	if err != nil {
+		return err
+	}
+
+	// do a verification check for the generated report
+	err = verifyReport(report, directory)
+	if err != nil {
+		return err
+	}
+
+	// cleanup after test
+	err = os.Remove(generatedFile)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getFilenames returns the filenames for a directory.
+func getFilenames(directory string) ([]string, error) {
+	var files []string
+	filepath.Walk(directory, func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// if not a directory, append to files
+		if !info.IsDir() {
+			filename := filepath.Join(directory, info.Name())
+			files = append(files, filename)
+		}
+
+		return nil
+	})
+
+	return files, nil
+}
+
+// Verify compares the generated report and parsed issues using tree-sitter.
+func verifyReport(report types.AnalysisReport, directory string) error {
+	var parsedIssues ParsedIssues
+
+	// get filenames
+	files, err := getFilenames(directory)
+	if err != nil {
+		return err
+	}
+
+	parser := sitter.NewParser()
+
+	// walk through each file and get issues
+	for _, filename := range files {
+		// get language
+		lang, err := getLanguage(filename)
+		if err != nil {
+			return err
+		}
+		parser.SetLanguage(lang)
+
+		// read report
+		content, err := os.ReadFile(filename)
+		if err != nil {
+			return err
+		}
+
+		// generate tree
+		ctx := context.Background()
+		tree, err := parser.ParseCtx(ctx, nil, content)
+		if err != nil {
+			return err
+		}
+
+		// create a query for fetching comments
+		queryStr := "(comment) @comment"
+		query, err := sitter.NewQuery([]byte(queryStr), lang)
+		if err != nil {
+			return err
+		}
+
+		// execute query on root node
+		qc := sitter.NewQueryCursor()
+		n := tree.RootNode()
+		qc.Exec(query, n)
+		defer qc.Close()
```

Comment:
This differ should be after the variable qc is initialized. It's deferred anyways.